internal/utils.go

```diff
@@ -0,0 +1,56 @@
+package internal
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/codecrafters-io/tester-utils/logger"
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+// MoveGitToTemp moves the system git binary to a temporary directory
+func MoveGitToTemp(harness *test_case_harness.TestCaseHarness, logger *logger.Logger) {
+	oldGitPath, err := exec.LookPath("git")
+	if err != nil {
+		panic(fmt.Sprintf("CodeCrafters Internal Error: git executable not found: %v", err))
+	}
+	oldGitDir := path.Dir(oldGitPath)
+
+	tmpGitDir, err := os.MkdirTemp("/tmp", "git-*")
+	if err != nil {
+		panic(fmt.Sprintf("CodeCrafters Internal Error: create tmp git directory failed: %v", err))
+	}
+	tmpGitPath := path.Join(tmpGitDir, "git")
+
+	command := fmt.Sprintf("sudo mv %s %s", oldGitPath, tmpGitDir)
+	moveCmd := exec.Command("sh", "-c", command)
+	moveCmd.Stdout = os.Stdout
+	moveCmd.Stderr = os.Stderr
+	if err := moveCmd.Run(); err != nil {
+		os.RemoveAll(tmpGitDir)
+		panic(fmt.Sprintf("CodeCrafters Internal Error: mv git to tmp directory failed: %v", err))
+	}
+
+	// Register teardown function to automatically restore git
+	harness.RegisterTeardownFunc(func() { restoreGit(tmpGitPath, oldGitDir) })
```

Comment:
```suggestion
	harness.RegisterTeardownFunc(func() { restoreGit(tmpGitPath, oldGitPath) })
```

It's confusing when the parameter name is "originalPath" and we're passing in the directory instead.